### ISSUE TITLE
how to limit the memory usage of GPU like TensorFlow?

### ISSUE THREAD

**Issue by Dinxin** [chainer/chainer#5896]
In TensorFlow, we can limit the memory usage of GPU like:

gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=0.7)
sess = tf.Session(config=tf.ConfigProto(gpu_options=gpu_options)) 

But, in chainer, what should we do if we want to limit the memory of GPU?

**Comment by jinjiren** [comment]
I think in chainer the size of GPU memory being used is **dependent on your model and batch size**, i.e., if you use a small model, it will only use a small amount of memory (like several tens or hundreds MBs).
On the other hand, by default TensorFlow will occupy almost all of your GPU memory no matter the size of your model. That's why it needs to have an option to set the fraction of GPU memory it can use.

In short, I think this function is not necessary for chainer. If your model costs a lot of memory in chainer, it means that you need to make your model small or decrease the batch size.

**Comment by kmaehashi** [comment]
We'll introduce this feature in CuPy v7.
https://docs-cupy.chainer.org/en/latest/reference/memory.html#limiting-gpu-memory-usage
